docs/API.md

```diff
@@ -74,6 +74,41 @@ Captures, also known as capturing groups, extract and store parts of the matched
 > [!NOTE]
 > TS Regex Builder does not have a construct for non-capturing groups. Such groups are implicitly added when required. E.g., `zeroOrMore(["abc"])` is encoded as `(?:abc)+`.
 
+### `backreference()`
+
+```ts
+function backreference(
+  groupNumber: number
+): Backreference
+```
+
+Regex syntax: `\1 \2`.
+
+A backreference is a way to match the same text as previously matched by a capturing group.
+
+### `namedCapture()`
+
+```ts
+function namedCapture(
+  sequence: RegexSequence
+): NamedCapture
+```
+
+Regex syntax: `(...)`.
+
+A named capturing group is a capturing group that give a name to the group. The group's matching result can later be identified by this name.
+
+### `namedBackreference()`
```

Comment:
Swift Regex Builder uses following convention for named captures/references:
```swift
let kind = Reference(Substring.self)

let regex = Capture(as: kind) {
  ChoiceOf {
    "CREDIT"
    "DEBIT"
  }
}
```

see: https://github.com/apple/swift-evolution/blob/main/proposals/0351-regex-builder.md#reference

It has a nice feature of connecting reference straight to capturing group, instead of forcing user to repeat the name twice, once in `capture` then in `backreference`.

However, in such case dropping "back" prefix seems beneficial, as reference becomes "backreference" only when added to regular expression. Until it's applied to the previous part of the express ("back"), is more of reference.